tasks/main.yml

```diff
@@ -40,14 +40,26 @@
 
 - name: setup agent repo reference
   yum_repository:
-    baseurl: "https://download.newrelic.com/infrastructure_agent/linux/yum/el/{{ (ansible_distribution == 'Amazon') | ternary('6', nrinfragent_os_version) }}/x86_64"
+    baseurl: "https://download.newrelic.com/infrastructure_agent/linux/yum/el/{{ (ansible_service_mgr == 'upstart') | ternary('6', '7') }}/x86_64"
     gpgcheck: "yes"
     gpgkey: https://download.newrelic.com/infrastructure_agent/gpg/newrelic-infra.gpg
     name: 'newrelic-infra'
     repo_gpgcheck: "{{ (nrinfragent_os_version == '5') | ternary('no', 'yes') }}"
     state: present
     description: New Relic Infrastructure
-  when: nrinfragent_os_name|lower == 'redhat'
+  when: nrinfragent_os_name|lower == 'redhat' and ansible_distribution|lower == 'amazon'
```

Comment:
you can leave this one only with `ansible_distribution|lower == 'amazon'` because if it's `amazon` the os will be `redhat`